app/validators/blacklisted_email_validator.rb

```diff
@@ -13,6 +13,7 @@ def blocked_email?(value)
 
   def on_blacklist?(value)
     return false if Rails.configuration.x.email_domains_blacklist.blank?
+    return true if BlacklistedEmailDomain.block?(value)
```

Comment:
You put the check _after_ `return false if Rails.configuration.x.email_domains_blacklist.blank?`. It means it does not check the black list on the database will not be checked if the environment variable is empty. I guess that is not what you expect.